Source/Core/Gen3/Searchers/WildSearcher3.cpp

```diff
@@ -97,6 +98,21 @@ std::vector<WildState> WildSearcher3::search(u8 hp, u8 atk, u8 def, u8 spa, u8 s
         return states;
     }
 
+    bool isRSESafariLocation = encounterArea.isRSESafariZone() && isRSEVersion;
```

Comment:
`bool rseSafariRockSmash = encounterArea.isRseSafariZone() && rse && encounter == Encounter::RockSmash;`

Removes needs for the switch and stuff

Reply:
fishing will need this too, can i leave this? or do i add another bool for fishing?

Reply:
> Do they do similar things to the RNG?

yes, basically the same thing

Reply:
I changed it as requested, we'll discuss this in later fishing pr!